blocks/rich-text/index.js

```diff
@@ -740,7 +740,9 @@ export class RichText extends Component {
 			!! this.editor &&
 			this.props.tagName === prevProps.tagName &&
 			this.props.value !== prevProps.value &&
-			this.props.value !== this.savedContent
+			this.props.value !== this.savedContent &&
+			! isEqual( this.props.value, prevProps.value ) &&
+			! isEqual( this.props.value, this.savedContent )
```

Comment:
This is the real fix. I removed this as part of #4955 but it's necessary to avoid unnecessarily content updates. I don't see any performance loss in adding this and it could even improve performance because we don't use those TinyMCE functions `setBookmark` and `setContent` constantly

Reply:
@youknowriad I feel like we removed and added this a few times I the last year now. :) Maybe worth leaving a comment?

Reply:
(I was surprised to see these lines again.)

Reply:
Good call 👍